src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -208,12 +208,15 @@ private unsafe static PEReader OpenPEFile(string filePath, out MemoryMappedViewA
             // well for us since it can make IL access very slow (call to OS for each method IL query). We will map the file
             // ourselves to get the desired performance characteristics reliably.
 
+            FileStream fileStream = null;
             MemoryMappedFile mappedFile = null;
             MemoryMappedViewAccessor accessor = null;
-
             try
             {
-                mappedFile = MemoryMappedFile.CreateFromFile(filePath, FileMode.Open, null, 0, MemoryMappedFileAccess.Read);
+                // Create stream because CreateFromFile(string, ...) uses FileShare.None which is too strict
+                fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read);
```

Comment:
We should open the stream with `useAsync: false` to avoid unnecessary allocations of ThreadPoolBoundHandle.